Sources/ListAdapter/AnyListSourcesBuilder.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  AnyListSourcesBuilder.swift
+//  ListKit
+//
+//  Created by Frain on 2022/8/17.
+//
+
+// swiftlint:disable line_length function_parameter_count
+
+public typealias AnyListSources<View: ListView> = ListAdaptation<AnySources, View>
+public typealias AnyTableSources = AnyListSources<TableView>
+public typealias AnyCollectionSources = AnyListSources<CollectionView>
+
+public extension ListAdaptation where Source == AnySources {
+    init<Source: ListAdapter>(
+        options: ListOptions = .init(),
+        @AnyListSourcesBuilder<View> content: () -> Source
+    ) where Source.View == View {
+        self.init(content(), options: options)
+    }
+}
+
+@resultBuilder
+public struct AnyListSourcesBuilder<View: ListView> {
+    public static func buildIf<S: ListAdapter>(_ content: S?) -> S? where S.View == View {
+        content
+    }
+
+    public static func buildEither<TrueContent, FalseContent>(
```

Comment:
Should we use `buildEither<TrueContent: ListAdapter, FalseContent: ListAdapter>` to make where clause more succinct?